packages/@react-stately/form/src/useFormValidationState.ts

```diff
@@ -82,8 +82,8 @@ function useFormValidationStateImpl<T>(props: FormValidationProps<T>): FormValid
   }
 
   // If the isInvalid prop is true, update validation result in realtime (controlled).
-  let controlledError: ValidationResult | null = isInvalid ? {
-    isInvalid: true,
+  let controlledError: ValidationResult | null = isInvalid !== undefined ? {
```

Comment:
Perhaps should be 
```suggestion
  let controlledError: ValidationResult | null = isInvalid != null ? {
```
just in case someone provides null to `isInvalid` expecting it to be equivalent to undef?

Reply:
I'm not sure about this one: 

1.  `useFormValidationState` doesn't get `null` as type for `isInvalid`
https://github.com/adobe/react-spectrum/blob/89fa370bd1bb346be643c9ab2a0d79fab857e4d7/packages/%40react-types/shared/src/inputs.d.ts#L25
2. `useControlledState` only interprets `undefined` as uncontrolled state 
https://github.com/adobe/react-spectrum/blob/89fa370bd1bb346be643c9ab2a0d79fab857e4d7/packages/%40react-stately/utils/src/useControlledState.ts#L21
3. I think it is quite common to interpret only `undefined` as uncontrolled value.

Reply:
correct, `isInvalid` should only be controlled (`true` or `false`, there's no other option) or uncontrolled (`undefined`)